src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -24,32 +24,214 @@
+    /**
+     * Fake correct REQUEST.
+     */
+    public static final RqFake REQUEST = new RqFake("GET", "/?code=200");
+
+    /**
+     * Mocked member profile json object.
+     */
+    @Mock
+    private transient MemberProfileJson memberprofilejson;
+
+    /**
+     * Test set up.
+     * @throws Exception If some problem insidemo
+     */
+    @Before
+    public void setUp() throws Exception {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUser.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+    }
 
     /**
      * PsLinkedin can login.
      * @throws IOException If some problem inside
-     * @todo #12:30min/DEV This class not implemented yet, but has to be.
-     *  Please implement it. Mocked OAuth server is one of the possible
-     *  solutions. In this case, we could use some library like WireMock
-     *  (see http://www.wiremock.org), which allows make stub for web
-     *  service. We'll need to make some changes in PsLinkedin,
-     *  that allow to change API url, and use local stubbed http
-     *  server during unit test.
-     */
-    @Ignore
+     */
+    @Test
+    public void successfulEnter() throws IOException {
+        new PsLinkedin(this.memberprofilejson)
+            .enter(REQUEST);
+    }
+
+    /**
+     * The authorization code is not provided with the REQUEST.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void codeIsNotProvided() throws IOException {
+        try {
+            new PsLinkedin(this.memberprofilejson)
+              .enter(new RqFake("POST", "/?someCode=codeIsNotProvided"));
+        } catch (final IllegalArgumentException ex) {
+            MatcherAssert.assertThat(
+                ex.getMessage(),
+                Matchers.is("code is not provided")
+            );
+        }
+    }
+
+    /**
+     * Invalid user first name filed in json.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void invalidFirstNameField() throws IOException {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidFirstName.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        // @checkstyle MultipleStringLiteralsCheck (1 line)
```

Comment:
@ikhvostenkov this is not required here (the following line doesn't have any strings)